src/Angular.js

```diff
@@ -840,43 +842,48 @@ function copy(source, destination) {
 
     stackSource.push(source);
     stackDest.push(destination);
-    return copyRecurse(source, destination);
+    return copyRecurse(source, destination, currentDepth);
   }
 
-  return copyElement(source);
+  return copyElement(source, currentDepth);
+
+  function copyRecurse(source, destination, currentDepth) {
+    currentDepth++;
+    if (maxDepth !== -1 && currentDepth > maxDepth) {
+      return typeof source;
```

Comment:
we are already using `...` for cyclic references so I think that we should do the same for this.